regression/goto-analyzer/constant_propagation_rounding_mode/main.c

```diff
@@ -0,0 +1,12 @@
+#include <assert.h>
+
+int main(void)
+{
+  __CPROVER_rounding_mode = 0;
```

Comment:
Yes, but the values for them are implementation defined and I can't use fesetround because it doesn't have a visible implementation in goto-analyzer. I could add a couple of defines like `#define FE_TONEAREST 0` though if you think that'd improve readability.

Reply:
fesetround should work so you can use the macros.  Redefining them will likely cause problems.

Reply:
`fesetround` doesn't work for me; If I call it with anything all the `__CPROVER` variables just become top immediately.

Reply:
This sounds like a bug that should be fixed rather than working around.  There are a couple of possible causes:

1. the libraries are not loaded in goto-analyze.
2. they are but the domain is not handling them correctly.

IIRC --show-goto-program should be after program loading and thus make 1 obvious.  --show should show the effect of analysis and so should identify 2.

Reply:
Do you get the same issue with both Makefile and CMake based builds? Probably goto-analyzer needs to link against the ansi-c library/dependency if it's not already doing so?

Reply:
It's pretty basic, I think: `link_to_library(goto_model, ui_message_handler);` is `#if 0`'ed.